### ISSUE TITLE
v2 database key `wt_media_file_m_file_multimedia_file_refn_index` is too long

### ISSUE THREAD

**Issue by Art4** [fisharebest/webtrees#2233]
Hi,

I'm trying to setup a webtrees environment for helping contributing (#1794). I'm using PHP 7.2.5 with MariaDB 10.1.32, webtrees at master branch (49902c73c8994709c11ae294a71fdfb0628774d3) and running the Setup wizard for webtrees. Then this error occured:

```
An unexpected database error occurred.

SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes (SQL: alter table `wt_media_file` add index `wt_media_file_m_file_multimedia_file_refn_index`(`m_file`, `multimedia_file_refn`))

#0 vendor\illuminate\database\Connection.php(624): Illuminate\Database\Connection->runQueryCallback('alter table `wt...', Array, Object(Closure))
#1 vendor\illuminate\database\Connection.php(459): Illuminate\Database\Connection->run('alter table `wt...', Array, Object(Closure))
#2 vendor\illuminate\database\Schema\Blueprint.php(97): Illuminate\Database\Connection->statement('alter table `wt...')
#3 vendor\illuminate\database\Schema\Builder.php(264): Illuminate\Database\Schema\Blueprint->build(Object(Illuminate\Database\MySqlConnection), Object(Illuminate\Database\Schema\Grammars\MySqlGrammar))
#4 vendor\illuminate\database\Schema\Builder.php(165): Illuminate\Database\Schema\Builder->build(Object(Illuminate\Database\Schema\Blueprint))
#5 app\Schema\Migration37.php(60): Illuminate\Database\Schema\Builder->create('media_file', Object(Closure))
#6 app\Database.php(211): Fisharebest\Webtrees\Schema\Migration37->upgrade()
#7 app\Http\Controllers\SetupController.php(288): Fisharebest\Webtrees\Database::updateSchema('\\Fisharebest\\We...', 'WT_SCHEMA_VERSI...', 42)
#8 app\Http\Controllers\SetupController.php(131): Fisharebest\Webtrees\Http\Controllers\SetupController->createConfigFile(Array)
#9 index.php(95): Fisharebest\Webtrees\Http\Controllers\SetupController->setup(Object(Symfony\Component\HttpFoundation\Request))
#10 {main}
```

**Comment by fisharebest** [comment]
Your database is set to use the `COMPACT` row format.

In the migration scripts, we select the `COMPRESSES` row format.
See https://github.com/fisharebest/webtrees/blob/master/app/Schema/Migration37.php#L59

This change should allow us to index columns longer than 255 characters.

I tested it on MySQL.

But if we cannot change the row format in MariaDB, then I guess we must change this field to 255 characters.

It is mostly used to store long URLs as media files.

**Comment by sibador** [comment]
Hi,

I have same error. I'm using PHP 7.1.26 with MariaDB 10.1.38. 

For solve this error, I add this ligne in my.ini config file for MariaDB server
innodb_large_prefix = 1
innodb_file_format = barracuda
And I restart MariaDB server for take into account the change.

Second solution for solve this error.  I change the field to 255 characters in file app/Schem/Migration37.php
Replace
            $table->string('multimedia_file_refn', 512); // GEDCOM only allows 30 characters 
With
            $table->string('multimedia_file_refn', **255**); // GEDCOM only allows 30 characters 

This error is only for MariaDB version 10.1.X. I thinks is fixed with version 10.2.X.
